addon/components/au-time-picker.js

```diff
@@ -1,22 +1,59 @@
 import Component from '@glimmer/component';
 import { action } from '@ember/object';
-import { tracked } from '@glimmer/tracking';
 import { isPresent } from '@ember/utils';
+import { trackedReset } from 'tracked-toolbox';
 
 export default class AuTimePickerComponent extends Component {
-  @tracked hourValue = this.args.hours || 12;
-  @tracked minuteValue = this.args.minutes || 0;
-  @tracked secondValue =
-    isPresent(this.args.showSeconds) && !this.args.showSeconds
-      ? 0
-      : this.args.seconds || 0; // ignore 'seconds' argument if 'showSeconds' is false
-  @tracked keyCodes = [8, 9, 13, 33, 34, 37, 39, 46];
+  @trackedReset({
+    memo: 'args.hours',
+    update() {
+      if (this.args.hours) {
```

Comment:
I wonder, won't this return `12` if the parent passes in `0` as the hour value?

Reply:
Omg yes :facepalm:. I just added a test for that, and will fix this. I truly hate the truthy-falsy stuff.

Reply:
I propose the check `if (this.args.hours || this.args.hours == 0)` instead. This also makes "0" as a string to be a valid time value.

Reply:
I would have gone strict mode and simply not allowed something else than numbers (or undefined) and thrown assertion errors when the user did pass something invalid to the component. Strict but simpler to implement. But I don't want to hijack any more of your time so this is fine 😄.